lib/oop.js

```diff
@@ -23,7 +23,43 @@ const mixin = (target, source) => {
   Object.assign(target, mix);
 };
 
+const REF_TYPES = ['object', 'function'];
+
+// Deep freeze object structure
+//   instance - <Object>, target nested objects structure
+// Returns: <Object> - deep freezed instance
+//
+const deepFreeze = instance => {
+  Object.freeze(instance);
+  const fields = Object.getOwnPropertyNames(instance);
+  for (const fieldName of fields) {
+    const value = instance[fieldName];
+    if (value === null) continue;
```

Comment:
What if `value === undefined`? I think we should skip it too

Reply:
`null` is checked separatelly because `typeof null === 'object'`
`undefined` will be skipped [here](https://github.com/metarhia/common/pull/355/files#diff-3791507a15f8cb71dc1a4f8377ca25674ef714aa9bb9d86995aa186d1ae92925R38)

Reply:
I guess `value == null` should work well.